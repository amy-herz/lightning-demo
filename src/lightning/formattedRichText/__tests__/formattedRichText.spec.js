import { createElement } from 'lwc';
import Element from 'lightning/formattedRichText';

const createRichText = () => {
    const element = createElement('lightning-formatted-rich-text', {
        is: Element,
    });
    document.body.appendChild(element);
    return element;
};

describe('lightning-formatted-rich-text', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('default', () => {
        const element = createRichText();
        expect(element).toMatchSnapshot();
    });

    it('Plain text, no formatting', () => {
        const element = createRichText();
        element.value = 'This is plain text';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('Simple rich text', () => {
        const element = createRichText();
        element.value =
            '<h2>This</h2> <p>is <i>some</i> simple <b>rich</b> text</p>';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('Rich text with hyperlink', () => {
        const element = createRichText();
        element.value =
            'This is a link to <a href="www.salesforce.com">Salesforce</a>';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('Rich text with links', () => {
        const element = createRichText();
        element.value =
            '<ul><li><a href="http://www.google.com">www.google.com</a></li><li>www.salesforce.com</li><li>http://www.google.com</li><li>salesforce.com</li><li>email@richtext.com</li></ul>';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('Should not throw when value is set to null', () => {
        const element = createRichText();
        expect(() => {
            element.value = null;
        }).not.toThrow();
    });

    // Sanitization library is not accessible from here, so this is not currently useful
    /* it('Rich text with non-whitelisted content', () => {
        const element = createRichText();
        element.value = "SVGs are not whitelisted: <svg height=\"300\" width=\"300\"><rect height=\"100%\" width=\"100%\" fill=\"blue\" /></svg>";
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });*/
});
