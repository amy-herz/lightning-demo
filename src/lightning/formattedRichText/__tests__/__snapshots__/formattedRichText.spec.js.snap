// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lightning-formatted-rich-text Plain text, no formatting 1`] = `
<lightning-formatted-rich-text
  class="slds-rich-text-editor__output"
>
  #shadow-root(open)
    <span>
      This is plain text
    </span>
</lightning-formatted-rich-text>
`;

exports[`lightning-formatted-rich-text Rich text with hyperlink 1`] = `
<lightning-formatted-rich-text
  class="slds-rich-text-editor__output"
>
  #shadow-root(open)
    <span>
      This is a link to &lt;a href="www.salesforce.com"&gt;Salesforce&lt;/a&gt;
    </span>
</lightning-formatted-rich-text>
`;

exports[`lightning-formatted-rich-text Rich text with links 1`] = `
<lightning-formatted-rich-text
  class="slds-rich-text-editor__output"
>
  #shadow-root(open)
    <span>
      &lt;ul&gt;&lt;li&gt;&lt;a href="http://www.google.com"&gt;www.google.com&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;a href="http://www.salesforce.com" target="_blank" rel="noopener"&gt;www.salesforce.com&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;a href="http://www.google.com" target="_blank" rel="noopener"&gt;http://www.google.com&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;a href="http://salesforce.com" target="_blank" rel="noopener"&gt;salesforce.com&lt;/a&gt;&lt;/li&gt;&lt;li&gt;&lt;a href="mailto:email@richtext.com"&gt;email@richtext.com&lt;/a&gt;&lt;/li&gt;&lt;/ul&gt;
    </span>
</lightning-formatted-rich-text>
`;

exports[`lightning-formatted-rich-text Simple rich text 1`] = `
<lightning-formatted-rich-text
  class="slds-rich-text-editor__output"
>
  #shadow-root(open)
    <span>
      &lt;h2&gt;This&lt;/h2&gt; &lt;p&gt;is &lt;i&gt;some&lt;/i&gt; simple &lt;b&gt;rich&lt;/b&gt; text&lt;/p&gt;
    </span>
</lightning-formatted-rich-text>
`;

exports[`lightning-formatted-rich-text default 1`] = `
<lightning-formatted-rich-text
  class="slds-rich-text-editor__output"
>
  #shadow-root(open)
    <span>
      
    </span>
</lightning-formatted-rich-text>
`;
